src/op_utils.rs

```diff
@@ -73,6 +73,128 @@ pub fn int_atom<'a>(args: Node<'a>, op_name: &str) -> Result<&'a [u8], EvalErr>
     }
 }
 
+pub fn uint_atom<const SIZE: usize>(args: &Node, op_name: &str) -> Result<u64, EvalErr> {
+    let bytes = match args.atom() {
+        Some(a) => a,
+        _ => {
+            return args.err(&format!("{op_name} requires int arg"));
+        }
+    };
+
+    if bytes.is_empty() {
+        return Ok(0);
+    }
+
+    if (bytes[0] & 0x80) != 0 {
+        return args.err(&format!("{op_name} requires positive int arg"));
+    }
+
+    // strip leading zeros
+    let mut buf: &[u8] = bytes;
```

Comment:
It seems confusing to require parsing `SIZE` bytes in this function but first drop potentially an infinite number of zero bytes here and then require the rest to be `SIZE` length but i guess thats some part of the protocol which im missing?

Reply:
the `SIZE` parameter specifies the limit of the variable the number is being parsed into. 8 bytes for `u64`, 4 bytes for `u32`. Leading zeroes do not contribute to the magnitude of the value, so they are ignored. This is part of how we interpret byte buffers as integers in clvm